fork_choice_control/src/tasks.rs

```
@@ -554,20 +554,14 @@ impl<P: Preset, W> Run for ReconstructDataColumnSidecarsTask<P, W> {
             block_root,
         } = self;
 
-        let Ok(available_columns) = (0..store_snapshot.chain_config().number_of_columns)
-            .map(|index| {
+        let available_columns = (0..store_snapshot.chain_config().number_of_columns)
+            .filter_map(|index| {
                 let identifier = DataColumnIdentifier { block_root, index };
-                match store_snapshot.cached_data_column_sidecar_by_id(identifier) {
-                    Some(data_column_sidecar) => Ok(Some(data_column_sidecar)),
-                    None => storage.data_column_sidecar_by_id(identifier),
-                }
+                store_snapshot
+                    .cached_data_column_sidecar_by_id(identifier)
+                    .or_else(|| storage.data_column_sidecar_by_id(identifier).ok().flatten())
```

I'm not a fan of silent failures. At least log a `WARN` on failure.